catroid/src/main/java/org/catrobat/catroid/content/Sprite.java

```diff
@@ -716,32 +719,61 @@ public Color getEmbroideryThreadColor() {
 		return this.embroideryThreadColor;
 	}
 
-	public void replaceSpriteWithSprite(Sprite sprite) {
-		this.scriptList = sprite.scriptList;
-		this.lookList = sprite.lookList;
-		this.soundList = sprite.soundList;
-		this.nfcTagList = sprite.nfcTagList;
-		this.userVariables = sprite.userVariables;
-		this.userLists = sprite.userLists;
-		this.userDefinedBrickList = sprite.userDefinedBrickList;
-	}
-
-	public void mergeSprites(Sprite sprite) {
+	public Sprite(Sprite sprite, Scene destinationScene) throws IOException {
+		copyLooksAndSounds(sprite, destinationScene, false);
 		this.scriptList.addAll(sprite.scriptList);
+		this.nfcTagList.addAll(sprite.nfcTagList);
+		this.userVariables.addAll(sprite.userVariables);
+		this.userLists.addAll(sprite.userLists);
+		this.userDefinedBrickList.addAll(sprite.userDefinedBrickList);
+		this.look = sprite.look;
+		this.myOriginal = sprite;
+		this.name = sprite.getName();
+	}
+
+	private void copyLooksAndSounds(Sprite sprite, Scene destinationScene,
+			boolean setUniqueName) throws IOException {
+		File imageDirectory = new File(
+				destinationScene.getDirectory(),
+				Constants.IMAGE_DIRECTORY_NAME
+		);
+		File soundsDirectory = new File(
+				destinationScene.getDirectory(),
+				Constants.SOUND_DIRECTORY_NAME
+		);
+
+		for (LookData data : sprite.lookList) {
```

Comment:
extracting the following lines to a method would lead to unecassery casting and wouldn't remove a lot of duplicated code, (everything would have to be wrapped in an if/else)  the generification at this level would make the code a lot less readable in my opinion